### ISSUE TITLE
Complex{Int} with maximum negative imaginary part does not print correctly

### ISSUE THREAD

**Issue by ethomag** [JuliaLang/julia#33433]
```julia
julia> Complex{Int8}(0,-128)
0 - -128im
```
The problem is in base/complex.jl, `function show(io::IO, z::Complex)`:

```julia
    if signbit(i) && !isnan(i)
        i = -i
```
Since -(Int8(-128)) == -128

**Comment by lhnguyen-vn** [comment]
#33498

**Comment by JeffBezanson** [comment]
Fixed by #33498
